client/connect.go

```diff
@@ -158,8 +159,28 @@ func (cli *Client) GetRemoteAccessToken(ctx context.Context, req RemoteAccessTok
 	return &remoteAccessToken, nil
 }
 
+// Connect establishes a connection to a remote devices
+func (cli *Client) Connect(ctx context.Context, connections map[string][]RemoteAccessTarget) error {
+	eg, ctx := errgroup.WithContext(ctx)
+
+	for deviceID, targets := range connections {
+		deviceID := deviceID
+		targets := targets
+
+		eg.Go(func() error {
+			// only print on errors, allow other connection attempts to continue
```

Comment:
In this case, I don't see much point in using the `errgroup`. I would suggest using `wg := sync.WaitGroup{}` and for each process, just do `wg.Add(1)` and do `wg.Done()` when its goroutine is finished. Then it's good-old-plain-go-routines:
```go
go func () {
	defer wg.Done()

	if err := cli.ConnectOne(ctx, deviceID, targets); err != nil {
		fmt.Printf("error connecting to device %s: %v\n", deviceID, err)
	}
}
```

And at the end `wg.Wait()`.